ui/src/pages/version/HistoryDetails.js

```diff
@@ -39,6 +33,10 @@ const DeploymentStatus = ({
   } else if (status === "pending") {
     return <EuiHealth color="gray">Pending</EuiHealth>;
   }
+
+  if (deployment.error !== "") {
```

Comment:
i think its more consistent to join the above if else block by checking `status === failed'

Reply:
Apparently, there's a case when status is running or serving but deployment.error is not empty, that's why I put it in different condition branch.

Reply:
if the status is running or serving, wont it returned `return <EuiHealth color="default">Not Deployed</EuiHealth>;` at the above instead?

Reply:
Yes, correct. That's the reason I moved deployment.error validation after checking the status because status takes precedence over error.

Reply:
so this `if (deployment.error !== "")` will only be called when `status===failed`? or there are other status? if the status is running or serving it wont call this block

Reply:
> so this if (deployment.error !== "") will only be called when status===failed? or there are other status? if the status is running or serving it wont call this block

I think the only possible status values are these: https://github.com/caraml-dev/merlin/blob/e78908cf7a1e89fb37771a1cff25cbbfd748a04d/api/models/endpoint_status.go#L20
So it's only when an endpoint is in the `failed` or `terminated` state where the code execution enters this block. That said, should we return `<EuiHealth color="danger">Terminated</EuiHealth>` if the `deployment.error === "" && status==="terminated"`? 

> if the status is running or serving it wont call this block

I don't think we're expecting the deployment to have an error if it's not in the `failed` state if I understood correctly 🤔